pydicom/multival.py

```diff
@@ -52,8 +53,14 @@ def __init__(
         from pydicom.valuerep import DSfloat, DSdecimal, IS
 
         def DS_IS_constructor(x: _T) -> _ItemType:
-            return self.type_constructor(x) if x != '' else cast(_ItemType, x)
-
+            return (  # type: ignore[no-any-return]
+                self.type_constructor(  # type: ignore[has-type]
```

Comment:
Not sure how to avoid these warnings.

Reply:
The problem is the optional argument to `type_constructor`. You could define a Protocol, or just leave it. In 3.10 you can use `ParamSpec`, I think.

Reply:
I'll leave it for the time being (actually missed that comment before...).